### PR TITLE
Performance network

### PR DESCRIPTION
**Pull request by pillip8282**
This PR contains optimization of network stack. 
And remove unused code.

### CODE REVIEW

File: framework/src/wifi_manager/Kconfig [hunk 1]
```diff
@@ -101,5 +101,11 @@ config DISABLE_EXTERNAL_AUTOCONNECT
 	bool "Disable external autoconnect"
 	default n
 
+config ENABLE_DRIVER_IOCTL
```
Comment (Taejun-Kwon) [inline framework/src/wifi_manager/Kconfig:104]: I don't think this config is necessary.
1. if user doesn't throw ioctl with specific args, ioctl will not be called
2. File ops is not enough to control h/w, so framework should control driver through ioctl. Is it really possible to control h/w without any of ioctl???? open read write close are enough?
3. If you want restrict some kind of get status or set power only, you should rename this config to specific one. Current is too generic.

Comment (pillip8282) [reply framework/src/wifi_manager/Kconfig:104]: Thanks for your opinion, Here are answers to your requests. I think answer for 3 can cover your question.
Yes now  operations only get status and  set power which is not mandatory, require ioctl interface. 
So we can make ioctl optional. 
HAL interface of wi-fi has defined basic operations like init, deinit, start sta commands. 
However as requirements from product are added, there are needs to add additional features to wi-fi driver. 
It's difficult to add wi-fi HAL interface every time when requirements are added. So I made ioctl  to apply requirement rapidly.

File: framework/src/wifi_manager/wifi_manager_state.c [hunk 2]
```diff
@@ -462,11 +462,11 @@ wifi_manager_result_e _handler_on_connecting_state(wifimgr_msg_s *msg)
 wifi_manager_result_e _handler_on_connected_state(wifimgr_msg_s *msg)
 {
 	if (msg->event == WIFIMGR_CMD_DISCONNECT) {
-		dhcpc_close_ipaddr();
+		//dhcpc_close_ipaddr();
```
[the remark this answers is not in the document]
Comment (pillip8282) [reply framework/src/wifi_manager/wifi_manager_state.c]: Thanks I'll modify it

File: framework/src/wifi_manager/wifi_manager_state.c [hunk 3]
```diff
@@ -462,11 +462,11 @@ wifi_manager_result_e _handler_on_connecting_state(wifimgr_msg_s *msg)
...
 		WIFIMGR_CHECK_RESULT(_wifimgr_disconnect_ap(), (TAG, "critical error\n"), WIFI_MANAGER_FAIL);
 		WIFIMGR_SET_STATE(WIFIMGR_STA_DISCONNECTING);
 	} else if (msg->event == WIFIMGR_EVT_STA_DISCONNECTED) {
-		dhcpc_close_ipaddr();
+		//dhcpc_close_ipaddr(); //2
```
[the remark this answers is not in the document]
Comment (pillip8282) [reply framework/src/wifi_manager/wifi_manager_state.c]: Thanks for your comment, I'll modify it

**Pull request closed without merging**